src/app/routines/myRoutines/routineCard/RoutineCard.tsx

```diff
@@ -5,29 +5,44 @@ import { useNavigate } from 'react-router-dom';
 import RoutineOptionsButton from './routineOptionsButton/RoutineOptionsButton';
 
 import './RoutineCard.scss';
-import { useMutation } from '@tanstack/react-query';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
 import { dislike, like } from '@/api/routineLikeApi';
 import AppAlert from '@/app/components/alerts/AppAlert';
-import { Routine } from '@/types/entities/Routine';
+import { PagedRoutine, Routine } from '@/types/entities/Routine';
 import LikeWithCount from '@/app/components/likeWithCount/LikeWithCount';
 import { AppAlertState } from '@/types/entities/AppAlert';
 
 const RoutineCard = ({ routine }: { routine: Routine }) => {
   const navigate = useNavigate();
+  const queryClient = useQueryClient();
   const [alertState, setAlertState] = useState<AppAlertState>({
     open: false,
     text: '',
     severity: 'error',
   });
 
-  const [likes, setLikes] = useState(routine.likesCount);
-  const [isLiked, setIsLiked] = useState(routine.isLikedByCurrentUser);
-
   const { mutate: likeRoutine, isPending: likeRoutinePending } = useMutation({
     mutationFn: (id: string) => like(id),
     onSuccess: () => {
-      setLikes((prev) => prev + 1);
-      setIsLiked(true);
+      queryClient.setQueryData(['routines'], (oldData: Routine[]) => {
```

Comment:
Ok, so in every place with likes, the approach is to find the routine that we modified and increment the like count.
Wouldn't it be more easier to invalidate the query and refetch all likes count:
```js
// Invalidate every query in the cache
queryClient.invalidateQueries()
// Invalidate every query with a key that starts with `todos`
queryClient.invalidateQueries({ queryKey: ['todos'] })

```

Reply:
You mean to refetch all routines every time you click a like button?

Reply:
Generally yes, but we can specify what to refetch based on queryKey, so we may fetch only the traget ones

Reply:
I can see 1 advantage of this approach and a few disadvantages.
Advantage: easier to implement probably
The main disadvantage: first you are sending request to add a like then you are sending request to refetch data - how can you ensure that you will receive an updated data?

Reply:
Ok, I see:
- From my point of view, we have cleaner and more maintainable code (otherwise we will need to refactor all of these methods in every place, when someone decides to change anything).
- About updated version of data - I believe it will be updated:
1. User likes or dislikes something
2. It is persisted in DB and we get a response 200
3. then we refetch the persisted data in DB, and get the updated value

Reply:
btw. I forgot to mention that I tried this approach and it didn't work 😆  maybe I did something wrong, I don't know but the approach I implemented now seems to me very reasonable and is very responsive. I don't know what you mean about refactoring all methods? This approach makes sense for likes because you don't make an additional call to db every time someone clicks a like but doesn't make sense to other things like updating workouts etc. so you will have these methods only here.

Reply:
okay, from what I see and your discussion, I like Hubert approach without refetching, although Id change the code structure, and extract the mapping methods, to make the code cleaner

Reply:
> okay, from what I see and your discussion, I like Hubert approach without refetching, although Id change the code structure, and extract the mapping methods, to make the code cleaner

that's actually something I would also like to do xd (as we discussed during the daily)